dvc/command/pipeline.py

```diff
@@ -167,8 +167,18 @@ def run(self):
 
         pipelines = self.repo.pipelines
         for p in pipelines:
+
+            # Get the stage with the lowest in-degree
+            p_edge = next(filter(lambda x: x[1] == 0, p.in_degree), None)
+            if not p_edge:
+                logger.error("Error finding an edge on the pipeline graph")
+            p_edge_name = p_edge[0]
             stages = networkx.get_node_attributes(p, "stage")
-            for stage in stages:
+
+            # Get the stage and get stages in order
+            nodes = networkx.dfs_postorder_nodes(p, p_edge_name)
```

Comment:
Would topological sort show us the same results?

Reply:
Thanks @efiop topological sort simplifies the logic

Reply:
@nvazquez I would check that it matches what `dvc repro --pipeline something.dvc` does.

Reply:
Here is the related piece of code https://github.com/iterative/dvc/blob/master/dvc/repo/reproduce.py#L89 . In short, it finds the pipeline for the specified target and then runs `dvc repro` on each end node in it, in no specific order(in terms of those end nodes). And understandably on sequential runs, the overlapping parts won't get executed again (they will be skipped as unchanged), so the resulting order will be quite odd and probably won't match what tsort will return. Though tsort might still give a good idea about what is going on. Your previous implementation would show those overlapping parts multiple times, which is not quite right.

Reply:
So maybe we let's adapt your previous approach to skip already executed stages? That would match `repro --pipeline` fully.